shared/team-building/service-tab-bar.native.tsx

```diff
@@ -0,0 +1,192 @@
+import {Props, IconProps} from './service-tab-bar'
+import {memoize} from '../util/memoize'
+
+const mapRange = (v: number, fromMin: number, fromMax: number, toMin: number, toMax: number) => {
+  return ((v - fromMin) / (fromMax - fromMin)) * (toMax - toMin) + toMin
+}
+
+export const labelHeight = 34
+
+const serviceMinWidthWhenSmall = (containerWidth: number) => {
+  const minWidth = 70
+  if (containerWidth <= minWidth) {
+    return minWidth
+  }
+  const p = containerWidth / minWidth // count that would fit onscreen at ideal size
+  let n = Math.floor(p) + 0.5
+  if (p % 1 < 0.5) {
+    n -= 1
+  }
+  // n = count that will fit onscreen at returned size
+  return containerWidth / n
+}
+
+const ServiceIcon = (props: IconProps) => {
+  const smallWidth = serviceMinWidthWhenSmall(Kb.NativeDimensions.get('window').width)
+  const bigWidth = Math.max(smallWidth, 92)
+  return (
+    <Kb.ClickableBox onClick={props.onClick}>
+      <Kb.Box2
+        direction="vertical"
+        centerChildren={true}
+        style={Styles.collapseStyles([
+          styles.serviceIconContainer,
+          {width: mapRange(props.labelPresence, 0, 1, smallWidth, bigWidth)},
+        ])}
+      >
+        <Kb.Icon
+          fontSize={18}
+          type={serviceIdToIconFont(props.service)}
+          style={Styles.collapseStyles([
+            styles.serviceIcon,
+            {color: props.isActive ? serviceIdToAccentColor(props.service) : inactiveServiceAccentColor},
+          ])}
+        />
+        <Kb.Box2
+          direction="vertical"
+          style={{
+            height: labelHeight * props.labelPresence,
+            opacity: props.labelPresence,
+            overflow: 'hidden',
+          }}
+        >
+          <Kb.Box2 direction="vertical" style={{height: labelHeight, width: 74}}>
+            <Kb.Text type="BodyTiny" center={true} lineClamp={2}>
+              {props.label}
+            </Kb.Text>
+          </Kb.Box2>
+        </Kb.Box2>
+        {!!props.showCount && props.count === null && (
+          <Kb.Icon
+            type="icon-progress-grey-animated"
+            color={Styles.globalColors.greyDark}
+            style={styles.pendingIcon}
+          />
+        )}
+        {!!props.showCount && props.count !== null && (
+          <Kb.Text type="BodyTinySemibold" style={styles.resultCount}>
+            {props.count && props.count === 11 ? '10+' : props.count}
+          </Kb.Text>
+        )}
+      </Kb.Box2>
+      <Kb.Box2
+        direction="horizontal"
+        fullWidth={true}
+        style={Styles.collapseStyles([
+          props.isActive ? styles.activeTabBar : styles.inactiveTabBar,
+          props.isActive && {backgroundColor: serviceIdToAccentColor(props.service)},
+        ])}
+      />
+    </Kb.ClickableBox>
+  )
+}
+
+const undefToNull = (n: number | undefined | null): number | null => (n === undefined ? null : n)
+
+const deriveOnScroll = memoize((deferClose, locked, setShowLabels) => () => {
```

Comment:
if you're memoizing a callback use `useCallback`

Reply:
Thanks. I'm unsure when to memoize / useCallback or not. Should onClose here also useCallback? Should I learn how to profile, like would there be enough signal in this function?

Reply:
you should def use the profiler. see: https://react-devtools-tutorial.now.sh/ for a nice tutorial
i'd lean on useCallback as its built for that purpose and its native to react